src/pages/RegisterVisit.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react';
+import DatePicker from 'react-datepicker';
+import { Box, Button, Flex } from '@chakra-ui/react';
+import SingleSelect from '../components/SingleSelect';
+import SingleDataPicker from '../components/SingleDataPicker';
+import { doctors, usersVisits } from '../utils/data';
+import { Doctor } from '../utils/types';
+
+export const labelSxStyle = { width: '120px' };
+export const FlexForLabelSxStyle = { alignContent: 'center' };
+export const mainBoxSxStyle = { margin: '135px 35px 35px 35px', width: '600px' };
+export default function RegisterVisit() {
+  const [databaseDoctors, setDatabaseDoctors] = useState<Doctor[]>(doctors);
+  const [selectedDoctor, setSelectedDoctor] = useState<string>('');
+  const [selectedUser, setSelectedUser] = useState<string>('');
+  const [startDate, setStartDate] = useState<Date>(new Date());
+
+  const handleExcludedTimes = (doctor: string) => {
+    const newArray = doctors.filter((el) => el.name === doctor);
+    if (newArray.length === 0) {
+      return;
+    } else {
+      return newArray[0].excludedDates;
+    }
+  };
+
+  const handleDateChange = (date: Date) => {
+    setStartDate(date);
+  };
+  const handleChangeSelectDoctor = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedDoctor(e.target.value);
+  };
+  const handleChangeSelectUser = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(e.target.value);
+  };
+  const handleSubmit = (e: React.MouseEvent<HTMLElement>) => {
+    e.preventDefault();
+
+    if (selectedDoctor.length === 0 || selectedUser.length === 0) return;
+
+    const newArrayDoctors = doctors.filter((el) => el.name === selectedDoctor);
+    const newArrayUserUser = usersVisits.filter((el) => el.name === selectedUser);
+    //jak ladniej zapisac push?tak to nie dziala
+    //const mwArr = [...newArrayDoctors[0].excludedDates, (startDate);]
+
+    newArrayDoctors[0].excludedDates.push(startDate);
```

Comment:
W React nie wolno robić push więc nie ma opcji żeby to ładniej zapisać

Reply:
nagle zaczelo dzialac, jest ok